### PR TITLE
Version 1.1.0

### PR DESCRIPTION
**Pull request by bulkware**
Azure discovery and metrics are not limited only to Application Insights anymore.

### CODE REVIEW

**Review by kompa3 — APPROVED** [review]
Some examples would be nice in readme, showing to to query eg. some application insight component metrics or some azure db metrics

**Comment by bulkware** [comment]
> Some examples would be nice in readme, showing to to query eg. some application insight component metrics or some azure db metrics

Examples were added in commit 14ae8af83c8b6ff171fb1fd523c3eb7eca271e0d.

**Comment by kompa3** [comment]
> > Some examples would be nice in readme, showing to to query eg. some application insight component metrics or some azure db metrics
> 
> Examples were added in commit [14ae8af](https://github.com/digiaiiris/zabbix-azure-monitoring/commit/14ae8af83c8b6ff171fb1fd523c3eb7eca271e0d).

Good!

File: etc/zabbix/zabbix_agentd.d/ic_azure.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# Microsoft Insights metric retrieval (configuration file, resource group, resource, timeshift, metric, statistic, timegrain)
-UserParameter=azure.insights.metric.timeshift[*],azure_insights_metric -c "$1" -g "$2" -r "$3" --timeshift "$7" "$4" "$5" "$6"
+# Azure metric retrieval (configuration file, resource group, resource, timeshift, metric, statistic, timegrain)
```
Comment (sapajune) [inline etc/zabbix/zabbix_agentd.d/ic_azure.conf]: New parameters are missing from list.

File: python/ic_azure/azure_discovery.py [hunk 2]
```diff
@@ -8,25 +8,30 @@
...
             resource_group
         )
-        resource_id += "/providers/microsoft.insights/components/{}".format(
+        resource_id += "/providers/{}/{}/{}".format(
+            provider_name,
+            resource_type,
             resource_name
         )
 
+        print resource_id
```
Comment (sapajune) [inline python/ic_azure/azure_discovery.py]: This should not be printed in discovery output.

File: README.md [hunk 3]
```diff
@@ -30,11 +30,11 @@ pip install https://github.com/digiaiiris/zabbix-azure-monitoring/releases/downl
 
 Item Syntax | Description | Units |
 ----------- | ----------- | ----- |
-azure.insights.discover_insights[configuration_file, resource_group, resource] | Discover metrics in Application Insights resources | {#METRIC_CATEGORY}, {#METRIC_NAME} |
+azure.discover[configuration_file, resource_group, provider_name, resource_type, resource] | Discover metrics from Azure resources | {#METRIC_CATEGORY}, {#METRIC_NAME} |
```
Comment (sapajune) [inline README.md:36]: Wouldn't it be nice to have resource parameters in the output so the discovered items get their arguments from discovery?

Also this discovers metrics for specific resource. Should there be discovery of resources of particular type?

**Review by sapajune — CHANGES_REQUESTED** [review]

File: etc/zabbix/zabbix_agentd.d/ic_azure.conf (hunk 1 above)
Comment (bulkware) [reply etc/zabbix/zabbix_agentd.d/ic_azure.conf]: I don't understand. Everything seems to be committed here:
https://github.com/digiaiiris/zabbix-azure-monitoring/blob/e0c32e3c1474da2d455d8e8592bc23d995c1cfea/etc/zabbix/zabbix_agentd.d/ic_azure.conf

Comment (bulkware) [reply etc/zabbix/zabbix_agentd.d/ic_azure.conf]: This was fixed in commit aa4284da257ce6fcb305b2bd90ebe741af84c58b.

File: python/ic_azure/azure_discovery.py (hunk 2 above)
Comment (bulkware) [reply python/ic_azure/azure_discovery.py]: Nice catch! This was for debugging purposes and was removed in commit 09df7ddb0e7d03dd24204998bab922a24ca136de.

**Review by sapajune — APPROVED** [review]

**Comment by sapajune** [comment]
Further improvements to discovery can be done separately.

**Review by Turilas83 — APPROVED** [review]

**Pull request merged**